src/Features/Core/Portable/Snippets/SnippetProviders/AbstractInlineStatementSnippetProvider.cs

```diff
@@ -83,7 +83,19 @@ private static bool TryGetInlineExpressionInfo(SyntaxToken targetToken, ISyntaxF
             syntaxFacts.IsExpressionStatement(parentNode?.Parent))
         {
             var expression = syntaxFacts.GetExpressionOfMemberAccessExpression(parentNode)!;
+
             var typeInfo = semanticModel.GetTypeInfo(expression, cancellationToken);
+            var symbolInfo = semanticModel.GetSymbolInfo(expression, cancellationToken);
+
+            // If we get the same symbol from symbol info as expression's type
+            // then we are dotting from a type name itself, e.g. `string.$$`.
+            // Inline statement snippets are not valid in this context
+            if (SymbolEqualityComparer.Default.Equals(symbolInfo.Symbol, typeInfo.Type))
```

Comment:
i'm wary about this due to the `Color Color` case.  Can we add a test or that.

also, to check for this case, i think you just need to check if the symbolInfo.Symbol is an INamedTypeSymbol (or just an ITypeSymbol).  You don't really need to check if they're equal.